superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/utils.ts

```diff
@@ -241,61 +172,23 @@ export const createHandleRemoveItem =
           ) => Record<string, FilterRemoval>)
         | Record<string, FilterRemoval>,
     ) => void,
-    setSaveAlertVisible: Function,
     setOrderedFilters: (
-      val: string[][] | ((prevState: string[][]) => string[][]),
+      val: string[] | ((prevState: string[]) => string[]),
     ) => void,
-    setFilterHierarchy: (
-      state:
-        | FilterHierarchy
-        | ((prevState: FilterHierarchy) => FilterHierarchy),
-    ) => void,
-    filterHierarchy: FilterHierarchy,
+    setSaveAlertVisible: Function,
   ) =>
   (filterId: string) => {
     const completeFilterRemoval = (filterId: string) => {
-      const buildNewFilterHierarchy = (hierarchy: FilterHierarchy) =>
-        hierarchy
-          .filter(nativeFilter => nativeFilter.id !== filterId)
-          .map(nativeFilter => {
-            const didRemoveParent = nativeFilter.parentId === filterId;
-            return didRemoveParent
-              ? { ...nativeFilter, parentId: null }
-              : nativeFilter;
-          });
       // the filter state will actually stick around in the form,
       // and the filterConfig/newFilterIds, but we use removedFilters
       // to mark it as removed.
       setRemovedFilters(removedFilters => ({
         ...removedFilters,
         [filterId]: { isPending: false },
       }));
-      // Remove the filter from the side tab and de-associate children
-      // in case we removed a parent.
-      setFilterHierarchy(prevFilterHierarchy =>
-        buildNewFilterHierarchy(prevFilterHierarchy),
+      setOrderedFilters((orderedFilters: string[]) =>
+        orderedFilters.filter(filter => !isEqual(filter, filterId)),
```

Comment:
```suggestion
        orderedFilters.filter(filter => filter !== filterId),
```
`isEqual` feels like a bit of an overkill for comparing 2 strings 🙂

Reply:
Oh man... my Java background gets me every time on this 😝. You should see me when I'm writing an `if` and thinking about Javascript, Java, and Python at the same time! It's hilarious 😆